tests/callbacks/test_TimerListener.cpp

```diff
@@ -0,0 +1,288 @@
+        EventRegexMatcher,
+        EnqueuedEventTimer<execution_space>
+    >;
+
+    static_assert(std::constructible_from<
+        region_timer_with_enqueued_event_timer_t,
+        std::regex, execution_space
+    >);
+}
+
+//! Timer that measures the time to launch, in addition to the time to execute the kernel.
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct EnqueuedEventTimerWithLatency
+{
+    //! Start the timer. The event must be on @ref exec.
+    template <EnqueuedEvent EventType>
+    void start(const EventType& event)
+    {
+        if(event.dev_id != dev_id)
+            return;
+
+        timer.start(exec);
+        timer_latency.start();
+    }
+
+    //! Stop the timer.
+    template <Event EventType>
+    void stop(const EventType&) {
+        timer_latency.stop();
+        timer.stop(exec);
+    }
+
+    template <typename Duration = Kokkos::utils::timer::milliseconds>
+    Duration duration() { return timer.template duration<Duration>(); }
+
+    template <typename Duration = Kokkos::utils::timer::milliseconds>
+    Duration latency() { return timer_latency.template duration<Duration>(); }
+
+    Exec exec;
+    uint32_t dev_id = Kokkos::Tools::Experimental::device_id(exec);
+    Kokkos::utils::timer::Timer<Exec> timer {};
+    Kokkos::utils::timer::Timer<void> timer_latency {};
+};
+
+template <MatcherFor<BeginParallelForEvent> MatcherType, Kokkos::utils::concepts::ExecutionSpace Exec>
+struct ParallelForTimerListenerWithLatency : public TimerListener<
+    EventBeginEndIdMatcher<ConjunctionMatcher<EventQueueMatcher<Exec>, MatcherType>, BeginParallelForEvent>,
+    EnqueuedEventTimerWithLatency<Exec>
+>
+{
+    using queue_matcher_t = EventQueueMatcher<Exec>;
+    using matcher_t       = EventBeginEndIdMatcher<ConjunctionMatcher<queue_matcher_t, MatcherType>, BeginParallelForEvent>;
+    using timer_t         = EnqueuedEventTimerWithLatency<Exec>;
+    using base_t          = TimerListener<matcher_t, timer_t>;
+
+    template <typename T>
+    ParallelForTimerListenerWithLatency(T&& matcher, const Exec& exec) : base_t(
+        matcher_t{.matcher = ConjunctionMatcher<queue_matcher_t, MatcherType>{queue_matcher_t{.exec = exec}, MatcherType{std::forward<T>(matcher)}}},
+        timer_t{.exec = exec}
+    ) {}
+};
+
+//! @test Check the parallel for timer listener when used with @ref EnqueuedEventTimerWithLatency.
+TEST_F(TimerListenerTest, with_latency)
+{
+    using parallel_for_timer_with_latency_t = ParallelForTimerListenerWithLatency<EventNameMatcher, execution_space>;
+
+    const auto par_for_timer = std::make_shared<parallel_for_timer_with_latency_t>(
+        "computation - level 0 - pfor", exec
+    );
+
+    Manager::register_listener(par_for_timer);
+
+    MyWorkload<execution_space>{}.execute(exec);
+
+    ASSERT_GT(par_for_timer->timer.duration(), Kokkos::utils::timer::milliseconds{0.});
+    ASSERT_GT(par_for_timer->timer.latency(),  Kokkos::utils::timer::milliseconds{0.});
```

Comment:
Let's add a check that latency was greater than 0 but smaller than the time for the call to `MyWorkload`.

Reply:
We'll need to do it in:
- #81